### PR TITLE
#160577585  Enable Users to opt in and out of notifications

### PR DESCRIPTION
**Pull request by imireallan**
## What does this PR do?
This PR allow users to subscribe and unsubscribe to email notifications

### Why this is important
Notifications give users to opportunity to know what they may have missed which may be important to them. However, forcing notifications on users may be seen as annoying. Users should have the ability to opt in and out of (or turn on and off) notifications.

### As a User
I want to be able to opt in and out of (or turn on and off) notifications.

### How should this be manually tested?**

- Log in to the app.
- Navigate to /settings route on the Navbar
- You should be taken to a page that displays the subscription switch

#### When the user subscribes to email notifications
image

#### When the user is unsubscribes from email notifications
image

## What are the relevant pivotal tracker stories?
The relevant pivotal story is [here](https://www.pivotaltracker.com/n/projects/2198535/stories/160577585)

### CODE REVIEW

**Review by gitaumoses4 — APPROVED** [review]
Looks good to me.

File: src/containers/Subscribe/index.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    componentWillReceiveProps = (nextProps) => {
+      const { subscriptionStatus, message, errors } = nextProps.sub;
+      const errorMessage = 'Something went wrong, Please try again';
+      this.setState({
+        notificationStatus: subscriptionStatus,
+      });
+      if (message.length !== 0) {
+        Materialize.toast({ html: message });
```
Comment (SrMoffat) [inline src/containers/Subscribe/index.js:25]: @imireallan ,
Good work so far! Even so, I am a bit concerned with this toast. I have tested this functionality and the toast disappears too quickly. I have the checked the default time for the toast is `375 milliseconds`. Maybe you can include a `outDuration` option on the toast and make it stay slightly longer like so:
```
if (message.length !== 0) {
        Materialize.toast({ html: message, outDuration: <SET_LONGER_TIME_HERE });
```

**Review by SrMoffat — APPROVED** [review]
@imireallan ,
Good work. It is good that you have used conditional rendering functionality for the `subscribe button`. I have seen it persists the user's subscription status as previously intended.

File: src/containers/Subscribe/index.js (hunk 1 above)
Comment (imireallan) [reply src/containers/Subscribe/index.js:25]: @SrMoffat the default onDuration time is longer enough for the user to see the message

**Pull request merged**